src/liberty/devCommands.ts

```diff
@@ -145,6 +145,40 @@ export async function startContainerDevMode(libProject?: LibertyProject | undefi
     }
 }
 
+export async function buildStarterProject( state?: any, libProject?: LibertyProject | undefined): Promise<void> {
+    var apiURL = `https://start.openliberty.io/api/start?a=${state.a}&b=${state.b}&e=${state.e}&g=${state.g}&j=${state.j}&m=${state.m}`;
+    const downloadStarterProject = async function(downloadLocation: string): Promise<void> {
+        axios({
+        method: "get",
+        url: apiURL,
+        responseType: "stream"
+        }).then( function (response){
+            response.data.pipe(fs.createWriteStream(downloadLocation))
+            .on("close", () => {
+                var unzip = require("unzip-stream");
+                fs.createReadStream(downloadLocation).pipe(unzip.Extract({ path: `${state.dir}/${state.a}`}));
+                fs.unlink(downloadLocation, async (err) => {
+                    const folderUri = vscode.Uri.file(state.dir);
```

Comment:
This seems to open the parent folder of the newly generated project rather than the newly generated project. For example, if I generate into my downloads, my entire downloads folder is opened in VS Code rather than just the `app-name` project I generated

Reply:
Fixed in fb3698a. Not sure if it is the flow we are looking for. I also used a timeout to refresh the file explore. I'll go back and use async/await, when I have more time.